projects/schematics/src/migrations/5_0/constructor-deprecations/data/constructor-deprecations.migration.ts

```diff
@@ -0,0 +1,4762 @@
+      },
+      {
+        className: 'OrderReturnRequestService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'RoutingService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'GlobalMessageService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'OrderDetailsService',
+        importPath: '@spartacus/order/components',
+      },
+      {
+        className: 'OrderReturnRequestFacade',
+        importPath: '@spartacus/order/root',
+      },
+      {
+        className: 'RoutingService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'GlobalMessageService',
+        importPath: '@spartacus/core',
+      },
+    ],
+  },
+  {
+    class: 'PageLayoutService',
+    importPath: '@spartacus/storefront',
+    deprecatedParams: [
+      {
+        className: 'CmsService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'LayoutConfig',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'BreakpointService',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'PageLayoutHandler',
+        importPath: '@spartacus/storefront',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'CmsService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'LayoutConfig',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'BreakpointService',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'PageLayoutHandler',
+        importPath: '@spartacus/storefront',
+      },
+    ],
+    addParams: [
+      {
+        className: 'CmsService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'LayoutConfig',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'BreakpointService',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'UnifiedInjector',
+        importPath: '@spartacus/core',
+      },
+    ],
+  },
+  {
+    class: 'PaymentMethodsComponent',
+    importPath: '@spartacus/storefront',
+    deprecatedParams: [
+      {
+        className: 'UserPaymentService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'TranslationService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'UserPaymentService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'TranslationService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'UserPaymentService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'TranslationService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'GlobalMessageService',
```

Comment:
this one is `optional` (`?`). Not sure whether it is breaking change.

Reply:
In this case, I think it's a change in the constructor, but it's not a breaking change.
I'll remove it.